biz.aQute.tester.junit-platform/src/aQute/tester/junit/platform/Activator.java

```diff
@@ -295,12 +297,26 @@ public void removedBundle(Bundle bundle, BundleEvent event, Bundle object) {
 		};
 		tracker.open();
 
+		if (Boolean.valueOf(context.getProperty("launch.services"))) {
+			try {
+				GogoCommandlet cmd = new GogoCommandlet();
+				Hashtable<String, Object> properties = new Hashtable<>();
+				properties.put("osgi.command.function", new String[] {
+					"runTests"
+				});
+				properties.put("osgi.command.scope", "tester");
+
+				trace("starting gogo commandlet");
+				context.registerService(Object.class, cmd, properties);
```

Comment:
Perhaps this should only be registered if in continuous mode? Otherwise there is no loop watching the queue. The test case sets continuous on to get the test to work I assume.

Reply:
> Perhaps this should only be registered if in continuous mode? Otherwise there is no loop watching the queue.

In auto, non-continuous mode, the tester can still it can still sit waiting if it doesn't find any test bundles. In theory, at least, the commandlet could still be used at this time. However if there are no test bundles in the environment it won't do much.

I'll change it to only register during continuous mode.

> The test case sets continuous on to get the test to work I assume.

Yes:
https://github.com/bndtools/bnd/blob/62dbaa2683a50c091393821995ad2a0ccb597bc7/biz.aQute.tester.test/test/aQute/tester/junit/platform/test/GogoShellTests.java#L76